format-delimited/src/main/java/io/cdap/plugin/format/delimited/input/DelimitedInputFormatProvider.java

```diff
@@ -111,25 +111,33 @@ static Schema detectSchema(DelimitedConfig conf, String delimiter,
     DataTypeDetectorStatusKeeper dataTypeDetectorStatusKeeper = new DataTypeDetectorStatusKeeper();
     for (InputFile inputFile : inputFiles) {
       String line;
-      String[] columnNames = null;
+
+      List<String> columnNamesList = null;
       String[] rowValue;
+
       try (BufferedReader reader = new BufferedReader(new InputStreamReader(inputFile.open()))) {
         for (int rowIndex = 0; rowIndex < conf.getSampleSize() && (line = reader.readLine()) != null; rowIndex++) {
+
           rowValue = line.split(delimiter, -1);
           if (rowIndex == 0) {
-            columnNames = DataTypeDetectorUtils.setColumnNames(line, conf.getSkipHeader(), conf.getEnableQuotedValues(),
-                                                               delimiter);
+
+            columnNamesList = DataTypeDetectorUtils.setColumnNames(line, conf.getSkipHeader(),
+                    conf.getEnableQuotedValues(), delimiter);
+            if (conf.getPathField() != null) {
```

Comment:
should not add it here. Add it to the list of fields right before the schema is created